enhancements/network/ovn-multi-external-gateway.md

```diff
@@ -0,0 +1,182 @@
+---
+title: ovn-multi-external-gateway
+authors:
+  - "@jordigilh"
+reviewers: 
+  - "@trozet"
+  - "@tssurya"
+  - "@fedepaol"
+approvers:
+  - "@trozet"
+api-approvers: 
+  - "@trozet"
+creation-date: 2023-02-06
+last-updated: 2023-02-08
+tracking-link: 
+  - https://issues.redhat.com/browse/SDN-2481
+---
+
+# OVN Multiple External Gateways
+
+## Summary
+
+When it comes to ingress traffic, load balancers and reverse proxies take care of redirecting traffic to one of potentially multiple destinations. Egress traffic is no different. Although it is already possible to use egress gateways, telco customers require the ability to dynamically configure one or more egress next hops per namespace via a common and secure API.
+
+OVN already supports the ability to configure multiple external gateways via pod and namespace annotations:
+
+* `k8s.ovn.org/routing-external-gws`
+* `k8s.ovn.org/external-gw-pod-ips`
+* `k8s.ovn.org/routing-namespaces`
+* `k8s.ovn.org/routing-network`
+* `k8s.ovn.org/bfd-enabled`
+
+However, to ensure only trusted deployments use these annotations the solution is to leverage on an admission webhook that filter which pods can use these annotations and in which namespaces.
+
+## Motivation
+
+The goal of this enhancement is to translate the need of an admission webhook to a custom resource controller.
+
+Using a CRD brings an additional security benefit as it is bound to an RBAC and it is possible for users to make changes to the egress gateway information without requiring changes to the deployment specs, thus making workloads deployment agnostic as they no longer contain the environment's gateway information embedded as annotations.
+
+### User Stories
+* As an OpenShift operator, I want to allow users the ability to define their custom network configuration for multiple external gateways without the need of a policy or rule applied by an admission controller to police over workloads and safeguard the external gateway configuration in a namespace.
+* As an OpenShift operator, I want to use RBAC to finetune access to the external gateway configuration. 
+* As an OpenShift user, I want to configure my namespace with multiple external gateways using a CR so that workload deployment and gateway information are no longer bundled together.
+
+### Goals
+
+- Migrate the current pod/namespace annotation based for the Multi External Gateway configuration to a new CRD.
+- Define a controller that reimplements the existing logic that handles the annotations but on the contents of the CRD.
+- Extend test coverage to include the new CRD use cases, matching the same use cases covered in the annotation tests.
+- Support for CRD configuration will be the only mechanism supported in OpenShift 4.14 and onwards. Annotations will still be supported in 4.13, however the CRD will be the only officially supported and documented metohd of configuration.
```

Comment:
```suggestion
- CRD configuration will be the only officially supported and documented mechanism in OpenShift >= 4.14. Annotations will be supported in OpenShift <= 4.13.
```
tried to reduce the clutter that was repeating some things. Hopefully I understood this correctly.